Source/Core/DolphinQt/Translation.cpp

```diff
@@ -254,6 +255,38 @@ static QStringList FindPossibleLanguageCodes(const QString& exact_language_code)
   return possible_language_codes;
 }
 
+static bool IsRTL(const QString& language_code)
+{
+  const QStringList components = language_code.split(u'_');
+
+  // ISO 15924 codes for modern scripts that are written right-to-left.
+  static const QSet<QString> rtl_scripts{
+      QStringLiteral("Adlm"), QStringLiteral("Arab"), QStringLiteral("Aran"),
+      QStringLiteral("Hebr"), QStringLiteral("Mand"), QStringLiteral("Mend"),
+      QStringLiteral("Nkoo"), QStringLiteral("Rohg"), QStringLiteral("Thaa"),
+      QStringLiteral("Samr"), QStringLiteral("Syrc"), QStringLiteral("Syre"),
+      QStringLiteral("Syrj"), QStringLiteral("Syrn"),
+  };
+
+  for (int i = 1; i < components.size(); ++i)
```

Comment:
I don't think theres a reason why this cannot be a range-based for-loop, can it?
Also, I wouldn't be surprised if there was an algorithm for this (altho I can't recall one from the top of my head)

Reply:
It starts at 1 instead of 0 because I wanted to skip the main language code, so making it into a range-based for loop would be awkward.

Reply:
The main language code should never actually be 4 in size, though...

Reply:
Oops, kinda missed that :D

Reply:
Instead of making this so hard on yourself you could just use
``QStringList::takeFirst()`` to take away the first component which you can then use in line ``205``.

Now you would be able to use a for-range loop or better yet just turn your ``QStringList`` into a ``QSet<QString>`` and use
``QSet::intersects`` (which will return true if at least one item matches the other)

Reply:
I made it use `QStringList::takeFirst` and a range-based for loop now.

However, I can't just use `QSet::intersects`, because I want to return either true or false as soon as I find a script code (four-letter component). `QSet::intersects` only lets me find matching script codes, not non-matching script codes.